Signal/src/ViewControllers/MediaTileViewController.swift

```diff
@@ -0,0 +1,560 @@
+                return defaultCell
+            }
+
+            guard let cell = self.collectionView?.dequeueReusableCell(withReuseIdentifier: MediaGalleryCell.reuseIdentifier, for: indexPath) as? MediaGalleryCell else {
+                owsFail("\(logTag) in \(#function) unexpected cell for indexPath: \(indexPath)")
+                return defaultCell
+            }
+
+            cell.configure(item: galleryItem, delegate: self)
+
+            return cell
+        }
+    }
+
+    // MARK: UICollectionViewDelegateFlowLayout
+
+    public func collectionView(_ collectionView: UICollectionView,
+                               layout collectionViewLayout: UICollectionViewLayout,
+                               referenceSizeForHeaderInSection section: Int) -> CGSize {
+
+        let kHeaderHeight: CGFloat = 50
+
+        switch section {
+        case kLoadOlderSectionIdx:
+            // Show "loading older..." iff there is still older data to be fetched
+            guard let mediaGalleryDataSource = self.mediaGalleryDataSource else {
+                owsFail("\(logTag) in \(#function) mediaGalleryDataSource was unexpectedly nil")
+                return CGSize.zero
+            }
+            return mediaGalleryDataSource.hasFetchedOldest ? CGSize.zero : CGSize(width: 0, height: 100)
+        case kLoadNewerSectionIdx:
+            // Show "loading newer..." iff there is still more recent data to be fetched
+            guard let mediaGalleryDataSource = self.mediaGalleryDataSource else {
+                owsFail("\(logTag) in \(#function) mediaGalleryDataSource was unexpectedly nil")
+                return CGSize.zero
+            }
+            return mediaGalleryDataSource.hasFetchedMostRecent ? CGSize.zero : CGSize(width: 0, height: 100)
+        default:
+            return CGSize(width: 0, height: kHeaderHeight)
+        }
+    }
+    // MARK: MediaGalleryDelegate
+
+    public func didTapCell(_ cell: MediaGalleryCell, item: MediaGalleryItem) {
+        Logger.debug("\(logTag) in \(#function)")
+        self.delegate?.mediaTileViewController(self, didTapView: cell.imageView, mediaGalleryItem: item)
+    }
+
+    // MARK: Lazy Loading
+
+    // This should be substantially larger than one screen size so we don't have to call it
+    // multiple times in a rapid succession, but not so large that loading more get's chopping
+    let kMediaTileViewLoadBatchSize: UInt = 80
+    var oldestLoadedItem: MediaGalleryItem? {
+        guard let oldestDate = galleryDates.first else {
+            return nil
+        }
+
+        return galleryItems[oldestDate]?.first
+    }
+
+    var mostRecentLoadedItem: MediaGalleryItem? {
+        guard let mostRecentDate = galleryDates.last else {
+            return nil
+        }
+
+        return galleryItems[mostRecentDate]?.last
+    }
+
+    var isFetchingMoreData: Bool = false
+
+    let kLoadOlderSectionIdx = 0
+    var kLoadNewerSectionIdx: Int {
```

Comment:
In Hungarian notation, `k` is used to indicate constants.  This isn't a constant.  

http://web.mst.edu/~cpp/common/hungarian.html

I've been trying to break myself of the habit of using hungarian notation in our codebase since it feels very un-swift-like and when Obj-C identifiers are auto-converted to Swift identifiers these prefixes tend to interfere.

Reply:
Whoops, fixed.